src/top_closure.ml

```diff
@@ -44,6 +52,38 @@ module Make(Keys : Keys) = struct
     match iter_elts elements ~temporarily_marked:Keys.empty with
     | Ok () -> Ok (List.rev !res)
     | Error elts -> Error elts
+
+  let top_closure_f ~key ~deps elements =
```

Comment:
Is it worth it to get rid of this duplicate with a functor over a monad? This is almost exactly the same as `top_closure`.

Reply:
Yh, that make sense

Reply:
done